server/src/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -264,6 +265,11 @@ public void scheduleRestartForVmsOnHost(final HostVO host, boolean investigate)
                 "Host [" + hostDesc + "] is down." + ((sb != null) ? sb.toString() : ""));
 
         for (VMInstanceVO vm : reorderedVMList) {
+            ServiceOfferingVO vmOffering = _serviceOfferingDao.findById(vm.getServiceOfferingId());
```

Comment:
I would recommend putting this changed inside the if scope in line 273. The additions are related to debug and if the debug is disabled, you can skip it.